rfcs/text/0000-extend-threat-group-software.md

```diff
@@ -0,0 +1,167 @@
+# 0000: Extend Threat Fieldset
+<!-- Leave this ID at 0000. The ECS team will assign a unique, contiguous RFC number upon merging the initial stage of this RFC. -->
+
+- Stage: **0 (strawperson)** <!-- Update to reflect target stage. See https://elastic.github.io/ecs/stages.html -->
+- Date: **TBD** <!-- The ECS team sets this date at merge time. This is the date of the latest stage advancement. -->
+
+<!--
+As you work on your RFC, use the "Stage N" comments to guide you in what you should focus on, for the stage you're targeting.
+Feel free to remove these comments as you go along.
+-->
+
+<!--
+Stage 0: Provide a high level summary of the premise of these changes. Briefly describe the nature, purpose, and impact of the changes. ~2-5 sentences.
+-->
+
+Currently the `threat` fieldset includes tactic, technique, and sub-techniques from the ATT&CK framework. ATT&CK also includes groups and software that we can easily add to the existing fieldset to include all of the ATT&CK framework. While these fields are directly referenced within the ATT&CK framework, they can also be used with other frameworks if `threat.framework` expands to use more than ATT&CK.
+
+## Fields
+
+<!--
+Stage 1: Describe at a high level how this change affects fields. Include new or updated yml field definitions for all of the essential fields in this draft. While not exhaustive, the fields documented here should be comprehensive enough to deeply evaluate the technical considerations of this change. The goal here is to validate the technical details for all essential fields and to provide a basis for adding experimental field definitions to the schema. Use GitHub code blocks with yml syntax formatting.
+-->
+
+<!--
+Stage 2: Add or update all remaining field definitions. The list should now be exhaustive. The goal here is to validate the technical details of all remaining fields and to provide a basis for releasing these field definitions as beta in the schema. Use GitHub code blocks with yml syntax formatting.
+-->
+
+### Proposed New Fields for Threat fieldset
+
+Field | Type | Example | Description
+--- | --- | --- | ---
+threat.software.id | keyword | S0552 | Software is a generic term for custom or commercial code, operating system utilities, open-source software, or other tools used to conduct behavior modeled in ATT&CK. This is the software identification number assigned by MITRE ATT&CK.
```

Comment:
Andy, given that we acknowledge ATT&CK may not be the only/permanent attack model, is there language we should add to indicate that these are just ATT&CK-specific examples? Maybe we could instead use something like "Software is a generic term for custom or commercial code...or other tools used to conduct behavior, commonly modeled using ATT&CK. This value is used for software identification, an example has been provided using ATT&CK."

Reply:
Yeah, absolutely I can do that. 

TBH, I think that we should drop all of this under `threat.framework.att&ck.*` but that'll be a breaking change that will be as impactful if we do it now or later...so I think we can spend more time now getting everything prepared then making sweeping changes when we're ready.

Reply:
Ah, so you're imagining a situation where there might be other threat.framework definitions - that strikes me as sound, though I couldn't name the top 3-4 alternatives that have a robust knowledge base.

Reply:
I completely agree...I'm just trying not to box us into ATT&CK this top level.

I think we're going to get to a point that we may need to next things under `threat.framework.att&ck` (or the like)
Example
```
threat.framework.att&ck.tactic.id
threat.framework.att&ck.tactic.name
threat.framework.att&ck.technique.id
threat.framework.att&ck.technique.name
threat.framework.att&ck.software.id
threat.framework.att&ck.software.name
...
```

I think that'll be a big change though.